lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java

```diff
@@ -156,6 +187,10 @@ public void collect(int doc) throws IOException {
           pqTop.score = score;
           pqTop = pq.updateTop();
           updateMinCompetitiveScore(scorer);
```

Comment:
We should propagate the global minimum score to the underlying scorer (`Scorer#setMinCompetitiveScore`)  too. The optimization above that compares the current doc with the global minimum score is a good one but propagating the global minimum score should be more effective since it will allow skipping hits directly in the scorer.

Reply:
@jimczi That breaks `testIndexSearcher.testCount` though

Reply:
I don't understand how it could break this test. We should propagate the min score only if the total hit threshold is reached and this test indexes 100 docs only while the default threshold is 10,000. So we should never skip hits in this test, am I missing something ?

Reply:
Hmm, yeah, not sure what happened. I beasted the tests again -- came in clean